core/src/main/java/feast/core/model/Store.java

```diff
@@ -119,26 +120,18 @@ public static Store fromProto(StoreProto.Store storeProto) throws IllegalArgumen
     }
   }
 
-  public List<Subscription> getSubscriptions() {
-    return Arrays.stream(subscriptions.split(","))
-        .map(this::convertStringToSubscription)
-        .collect(Collectors.toList());
-  }
-
-  private static String convertSubscriptionToString(Subscription sub) {
-    if (sub.getName().isEmpty() || sub.getProject().isEmpty()) {
-      throw new IllegalArgumentException(
-          String.format("Missing arguments in subscription string: %s", sub.toString()));
-    }
-    return String.format("%s:%s", sub.getProject(), sub.getName());
-  }
+  /**
+   * Returns a List of Subscriptions based on whether to include those with excluded flag.
+   * Currently, store's subscriptions string will store ALL subscriptions, but during retrieval for
+   * subscriptions, the exclude flag provided by this function will be used for filtering.
+   *
+   * @param exclude boolean flag to signify whether to return subscriptions which have excluded flag
+   * @return List of Subscription that based on exclusion filter
+   */
+  public List<Subscription> getSubscriptionsByStr(boolean exclude) {
```

Comment:
This method makes the caller think that they will pass a string. `ByStr` implies that. Also, this exclusion flag probably is introduced to provide backward compatibility, but it seems out of place. Ideally we would not be filtering the subscription list at this level, but we should have a means to apply that subscription list for detection, for example as you have done with `isSubscribedToFeatureSet`. 

What would the impact be of removing this field?